DataFormats/OnlineMetaData/test/onlineMetaDataRecord_t.cpp

```diff
@@ -0,0 +1,141 @@
+#include <cppunit/extensions/HelperMacros.h>
+#include "DataFormats/OnlineMetaData/interface/DCSRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineBeamSpotRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineLuminosityRecord.h"
+#include "DataFormats/OnlineMetaData/interface/OnlineMetaDataRaw.h"
+
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <sstream>
+#include <string.h>
+#include <vector>
+
+
+class TestOnlineMetaDataRecord: public CppUnit::TestFixture
+{
+  CPPUNIT_TEST_SUITE(TestOnlineMetaDataRecord);
+  CPPUNIT_TEST(testDCSRecord);
+  CPPUNIT_TEST(testOnlineBeamSpotRecord);
+  CPPUNIT_TEST(testOnlineLuminosityRecord);
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+
+  void setUp() override;
+  void tearDown() override {}
+
+  void testDCSRecord();
+  void testOnlineBeamSpotRecord();
+  void testOnlineLuminosityRecord();
+
+private:
+  const std::string dumpFileName = "dump_run000001_event00020137_fed1022.txt";
+  std::vector<uint32_t> data;
+  onlineMetaData::Data_v1 const* onlineMetaData;
+  //OnlineMetaDataRecord onlineMetaDataRecord;
+};
+
+///registration of the test so that the runner can find it
+CPPUNIT_TEST_SUITE_REGISTRATION(TestOnlineMetaDataRecord);
+
+
+void TestOnlineMetaDataRecord::setUp() {
+
+  if ( data.empty() ) {
+    char* cmsswBase;
+    cmsswBase = getenv("CMSSW_BASE");
+    std::ostringstream dumpFileStr;
+    dumpFileStr << cmsswBase << "/src/DataFormats/OnlineMetaData/test/" << dumpFileName;
+
+    std::ifstream dumpFile(dumpFileStr.str().c_str());
+    uint32_t address;
+
+    std::string line, column;
+    while ( dumpFile.good() ) {
+      getline(dumpFile, line);
+      //std::cout << line << std::endl;
+      std::istringstream iss(line);
+      if ( ! (iss >> std::hex >> address >> column) ) continue;
+      for ( int i = 0; i < 4; ++i) {
+        uint32_t value;
+        if ( iss >> std::hex >> value ) {
+          data.push_back(value);
+        }
+      }
+    }
+    dumpFile.close();
+
+    CPPUNIT_ASSERT( data.size() == 36 );
+
+    onlineMetaData = reinterpret_cast<onlineMetaData::Data_v1 const*>(data.data());
+    //    onlineMetaDataRecord = OnlineMetaDataRecord((unsigned char*)data.data());
+    //    std::cout << onlineMetaDataRecord << std::endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant

Reply:
It's a left over. It will be removed.